.github/workflows/build.yml

```diff
@@ -1,38 +1,49 @@
 name: Build
-on: [pull_request]
+on: [pull_request, workflow_dispatch]
 jobs:
   cocoapods:
     name: CocoaPods
     runs-on: macOS-latest
     steps:
       - name: Check out repository
-        uses: actions/checkout@v1
-      - name: Use Xcode 12.3
-        run: sudo xcode-select -switch /Applications/Xcode_12.3.app
+        uses: actions/checkout@v2
+      - name: Use Xcode 12.4
+        run: sudo xcode-select -switch /Applications/Xcode_12.4.app
       - name: Install CocoaPod dependencies
         run: pod install
       - name: Run pod lib lint
-        run: pod lib lint --allow-warnings
+        run: pod lib lint
   carthage:
     name: Carthage
     runs-on: macOS-latest
     steps:
       - name: Check out repository
-        uses: actions/checkout@v1
-      - name: Use Xcode 12.3
-        run: sudo xcode-select -switch /Applications/Xcode_12.3.app
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Use Xcode 12.4
+        run: sudo xcode-select -switch /Applications/Xcode_12.4.app
       - name: Remove SPMTest
-        run: rm -rf SampleApps/SPMTest && git add . && git commit -m 'Remove SPMTest app to avoid Carthage timeout'
-      - name: Run carthage build
-        run: carthage build --no-skip-current
+        run: |
+          git checkout $GITHUB_HEAD_REF
+          rm -rf SampleApps/SPMTest
+          git add SampleApps/SPMTest
+          git commit -m 'Remove SPMTest app to avoid Carthage timeout'
+      - name: Use current branch
+        run: |
+          echo 'git "file://'$PWD'" "'$GITHUB_HEAD_REF'"' > SampleApps/CarthageTest/Cartfile
+      - name: Run carthage update
+        run: cd SampleApps/CarthageTest && carthage update
```

Comment:
This could be a separate PR - but maybe we can require our repo uses [Carthage 0.37.0](https://github.com/Carthage/Carthage/releases/tag/0.37.0), and we can get rid of that `carthage.sh` script (which is only used in the Rakefile at this point).